lib/inspec/file_provider.rb

```diff
@@ -0,0 +1,219 @@
+# encoding: utf-8
+require 'rubygems/package'
+require 'zlib'
+require 'zip'
+
+module Inspec
+  class FileProvider
+    def self.for_path(path)
+      if path.is_a?(Hash)
```

Comment:
feels like we want to have a resolve method in our provider and just iterate through the instances.